cocos/renderer/gfx-metal/MTLBuffer.mm

```diff
@@ -58,12 +49,17 @@ MTLResourceOptions toMTLResourseOption(GFXMemoryUsage usage)
     }
     
     if (_usage & GFXBufferUsage::VERTEX ||
-        _usage & GFXBufferUsage::INDEX ||
         _usage & GFXBufferUsage::UNIFORM)
     {
-        _mtlResourceOptions = toMTLResourseOption(info.memUsage);
-        if (_size > 0)
+//        for single-use data smaller than 4 KB, use setVertexBytes:length:atIndex: instead
+        if (_size < MINIMUMR_REQUIRED_SIZE_4KB)
```

Comment:
It is better to add the link for the optimization to describe why use small vertex buffer like this.